isc_roboteq/src/isc_roboteq.cpp

```diff
@@ -120,13 +126,30 @@ void Roboteq::driveCallBack(const geometry_msgs::msg::Twist::SharedPtr msg)
 
 void Roboteq::recieve(std::string result)
 {
+  roboteq_msgs::msg::EncoderCounts counts;
+  int count;
   if (result.empty()) 
   { 
     RCLCPP_ERROR(this->get_logger(), "%s","Failed to receive an echo from Roboteq:(");
   }
-  // put encoder parser and publisher here
-  // ...
-  // ...
+
+  //If encoders are present, check the recieved message to see if its an encoder message
+	if (has_encoders && result.substr(0, 3) == "CR=") {
```

Comment:
Consider wrapping this if block with a try catch block, as substr will throw `out_of_range` if for whatever reason you recived something other than the encoders. In theory this doesn't matter, but it's best to keep the node from dieing in case some edge case causes this to be an issue.

Reply:
I doubt this will become an issue unless encoder values come in in a different way than anticipated but its still a good idea to have so I implemented it